README
#==========================================#

PopojiCMS
Free Content Management System - Indonesia

Contact : info@popojicms.org

#==========================================#

A. Cara instalasi PopojiCMS :

   1. Extract file popojicms.v.x.x.x.zip di directory web Anda.
   2. Buatlah database baru yang nantinya sebagai tempat instalasi tabel-tabel.
   3. Melalui browser Anda, masuk ke alamat web dimana file popojicms.v.x.x.x.zip tadi diextract.
   4. Ikuti petunjuk instalasi dengan benar.
   5. Jika instalasi berhasil, hapuslah atau rename file install.php dan hapus README file ini dari directory web Anda.
   6. PopojiCMS siap untuk digunakan.

   Catatan :
   Jika diinstall pada localhost maka settingan,
   1. rewrite_module = on
   2. jika terjadi error "500 internal server error" (web telah di hosting), kemungkinan :
	1. karena pada file .htaccess belum ada baris code "RewriteBase /"
	solusi : tambahkan baris code RewriteBase / sebelum code RewriteEngine on

   3. jika terjadi error "The page isn't redirecting properly" atau "This webpage has a redirect loop" :
	1. coba periksa kembali apakah rewrite_module sudah on atau belum
	2. periksa apakah file .htaccess tercopy pada server local atau hosting dengan baik
	3. setelah itu clear cache browser Anda
				
   4. jika terdapat error yang lain, mungkin karena hasil extract file yang tidak sempurna, silahkan replace file-file yang error tersebut.
	
   5. untuk di hosting, lakukan perubahan user permission untuk folder po-upload menjadi 775 (po-content --> uploads).

B. Login backend PopojiCMS :
   1. Masuk ke alamat http://nama.web.anda/po-admin
   2. Masukkan data login sebagai berikut :
	Username : seperti yg telah diinputkan pada saat proses instalasi.
	Password : seperti yg telah diinputkan pada saat proses instalasi.


#==========================================#

Terima Kasih

#==========================================#

   1. Tuhan Yang Maha Esa
   2. Orang-orang yang berada di belakang PopojiCMS
   3. Aries sebagai pembuat template backend v.1.0.1 - v.1.1.1
   4. Aquincum sebagai pembuat template backend v.1.1.2 - v.1.2.2
   5. ProUI sebagai pembuat template backend v.1.2.3 - v.1.3.0
   6. Enews, Magazine, Andia, Brownie, Wiretree, Neon, Pressroom dan Canvas sebagai pembuat template frontend
   7. StructureCore Installation sebagai referensi modul instalasi
   8. Easy Menu Manager sebagai pembuat component menu manager
   9. FluentPDO, Bramus, Plates dan semua library php yang dipakai pada PopojiCMS
   10. Jquery, Bootstrap dan semua plugins jquery yang dipakai pada PopojiCMS


#==========================================#

Changelog PopojiCMS

#==========================================#

Build Version 2.0.1 (07 Agustus 2016) :
# Perbaikan pemanggilan library utama pada core.php
# Penambahan pilihan timezone pada proses instalasi.
# Penambahan requirement php extention fileinfo pada proses instalasi.
# Penambahan option penggunaan vQmod pada config.php
# Penambahan option timezone pada config.php
# Perbaikan publish date yang bermasalah dengan menyamakan waktu sistem dan database.
# Perbaikan modal window tinymce menjadi responsive.
# Perbaikan masalah pada component library dan filemanager.
# Penambahan rss feed pada frontend.
# Penambahan snipped post populer per kategori.
# Perbaikan engine dynamic menu.
# Perbaikan masalah pengiriman email notifikasi.
# Perbaikan plugin javascript time pada component post.
# Perbaikan component user pada masalah edit data.
# Penambahan option permalink pada url post.
# Penambahan info date dan hits pada component tabel post.
# Perbaikan design halaman login member.
# Penambahan halaman profile member.
# Penambahan halaman detail statistik visitor.

Build Version 2.0.0 (07 Juni 2016) :
# Perubahan keseluruhan engine dari native ke struktur MVC tetapi tetap dengan rasa native.
# Penambahan tag pada bagian frontend.
# Penambahan komentar sehingga sudah mendukung struktur tree (parent-child).
# Penambahan data konfigurasi persub pada bagian pengaturan engine.
# Pergantian wajah theme chingsy.
# Telah mendukung 100% engine multi bahasa.
# Galeri pada frontend sudah dibuat peralbum.
# Penambahan gambar galeri pada component post.
# Penambahan schedule post.
# Penambahan kategori sehingga sudah mendukung struktur tree (parent-child).
# Setiap post sudah mendukung multi kategori.
# Perbaikan sistem pencarian post pada frontend.
# Perbaikan component pustaka sehingga semua file yang diupload akan termanage dari sini.
# Penambahan konfigurasi pada engine theme.
# Penambahan istilah widget pada component.
# Perbaikan level pengguna yang lebih dinamis.
# Penambahan menu admin yang bisa diganti lewat component menu manager.
# Penambahan engine ternama vqmod.
# Support semua database yang bisa dijalankan dengan PDO.
# Penambahan dashboard member terpisah.
# Penambahan picture description pada post.

Build Version 1.3.0 (09 Februari 2015) :
# Tahun baru, wajah baru.
# Perbaikan engine menu yang terjadi error double class ketika digenerate.
# Menghilangkan fitur voice recognizer (sementara) yang tidak support lagi di chrome yang baru.
# Update engine filemanager terbaru sesuai update dari pihak ketiga.
# Penyesuain element html dengan W3C Standards.
# Rilis CoGen v.1.0.0 yaitu tool untuk meng-generate component otomatis tanpa perlu menulis script lagi.
# Penambahan social media meta tag sehingga setiap post akan lebih spesifik terbaca ketika share link di social media.
# Perbaikan fungsi extract file zip.
# Penambahan fast menu untuk tambah baru data di component-component.
# Penambahan permalink dan publish date pada component post dan pages.
# Perbaikan filtering post pada back-end sehingga semua user yang berlevel admin dapat melihat semua post dari user yang lain.
# Penambahan action reply pada component contact.
# Penambahan plugin jquery untuk progres form yang ada di login, register, activation dan recover.
# Penambahan tipe login password yaitu login dengan pattern seperti pada android.
# Penambahan image medium untuk setiap image yg diupload (po_upload/medium/medium_namaimage.extensi).
# Penambahan sitemap generator dibagian setting/pengaturan po-admin.
# Pembaharuan template back-end dan frond-end agar bisa dibuka cross browser (termasuk IE).
# Captcha di komentar sudah diganti dengan new reCaptcha dari google.
# Perbaikan maintenance template.
# Penambahan fitur headline dan publish time pada component post.
# Perbaikan pada editor tinymce.
# Perbaikan security yang ada pada tiap component.

Build Version 1.2.5 (12 Agustus 2014) :
# Perbaikan fitur theme editor menggunakan script Code Mirror, menjadikan editor theme lebih powerfull lagi.
# Penambahan helper pada theme editor, sehingga memudahkan developer untuk membuat themenya sendiri.
# Penambahan fitur Speech Recognition pada component post sehingga menulis tidak perlu lagi mengetikan keyboard tetapi cukup dengan berbicara pada microphone saja (Only Google Chrome).
# Penambahan user registrasi menjadikan PopojiCMS menjadi web multiuser yang dinamis sesuai dengan hak akses tiap user.
# Penambahan component user akses sehingga secara default hak akses terdiri dari admin, user dan member, yang nantinya bisa ditambahkan sendiri hak akses sesuai kebutuhan end user.
# Penambahan component user role dimana component ini dapat mengatur secara dinamis hak akses masing-masing content yang ada di halaman administrator sesuai dengan user akses tertentu.
# Perbaikan .htaccess dimana telah diatur slug url baru yang lebih SEO Friendly lagi, hal ini untuk menanggapi permintaan dari kebanyakan end user yang menginginkan slug url yang lebih baik.
# Perbaikan 2 theme default mengikuti slug url baru serta perbaikan fitur pagging yang lebih dinamis lagi.
# Perbaikan pada title, meta descriptions, meta keywords untuk hasil SEO yang lebih baik lagi.

Build Version 1.2.4 (7 Juni 2014) :
# Perbaikan pada menu engine, dimana sekarang sudah bisa support multi class sampai 3 class ke bawah.
# Perbaikan pada .htaccess baik pada frond-end ataupun pada back-end.
# Perbaikan setiap inputan pada frond-end tidak lagi menggunakan ajax tetapi melalui pemprosesan php murni, hal ini untuk mempermudah developing theme nantinya.
# Penambahan fitur pagging dimana sekarang bisa dipilih untuk pagging berupa kombinasi number atau hanya tombol prev dan next saja.
# Perbaikan penulisan coding pada komponen galeri dan user.
# Perbaikan pada komponen post dimana sekarang untuk user yang tidak memiliki hak akses admin hanya bisa melihat dan mengedit konten yang ditulis oleh dirinya sendiri.

Build Version 1.2.3 (1 April 2014) :
# Peleburan PopojiCMS menjadi versi 1 bahasa saja.
# Perubahan menyeluruh pada struktur theme backend dengan menggunakan theme ProUI.
# Penyempurnaan engine theme frond-end, selain tampilan baru, skinnya sudah bisa diganti-ganti.
# Mode maintenance yang bisa diatur lewat backend.
# Sistem file caching untuk mempercepat loading website.
# .httacess gzip mode untuk mengkompresi file web menjadi plain text untuk mempercepat loading website.
# Penggunaan natural bootstrap yang memudahkan developer bermain dengan designnya.
# Penambahan sistem notifikasi untuk komentar dan kontak yang masuk dan belum terbaca.
# Pada backend semua datatable telah menggunakan ajax-json sehingga data lebih cepat ditampilkan.
# Upload data untuk image post telah menggunakan filemanager 1 atap, sehingga lebih mudah dalam pengoprasiannya.
# Text editor dijadikan default newtinymce yang tampilannya sudah disesuaikan dengan design.
# Penggantian konten post dengan konten yang lebih baru.

Build Version 1.2.2 (27 Januari 2014) :
# Pergantian logo PopojiCMS baik favicon maupun logo pengenal diganti dengan logo yang baru.
# Pelucuran PopojiCMS versi 1 bahasa, slim and simple.
# Rilis PopojiCMSOAuth v.1.0, untuk facebook dan twitter connect.
# Penambahan multi delete item pada component : post, pages, category, tags, media, contact, comment, gallery.
# Penambahan javascript di file functions.js (po-admin/js/files/functions.js) untuk mendukung multi delete item.
# Penambahan function cuthighlight(), tgl_indo(), tgl_eng() di file po-library/po-function.php.
# Penambahan fitur lupa password akun pada halaman administrator.
# Penambahan fitur timezone pada component setting.
# Penambahan paging php pada frond-end sebagai option lain jquery paging sebelumnya.

Build Version 1.2.1 (04 Desember 2013) :
# Perbaikan theme editor pada versi sebelumnya yang ternyata terjadi kesalahan target link.
# Perbaikan edit user yang terjadi error pada script session_register().

Build Version 1.2.0 (07 November 2013) :
# Perubahan besar pada engine theme front-end, mempermudah implementasi template baru.
# Perbaikan security dan error pada versi sebelumnya.

Build Version 1.1.5 (14 Oktober 2013) :
# Perbaikan pada recaptcha di theme chingsy, diganti menggunakan theme clean karena terjadi error pada beberapa tampilan browser.
# Perbaikan pada script pencarian di theme chingsy.
# Pemindahan classmenu.php ke direktori library dan di rename menjadi po-classmenu.php.
# Penambahan editor baru yaitu newtinymce menggantikan editor lama cleditor.

Build Version 1.1.4 (28 September 2013) :
# Perbaikan theme pada instalation page.
# Penambahan inputan username dan password default pada saat instalasi.
# Perbaikan component post, karena terjadi error pada beberapa web hosting disaat input dan update artikel.
# Perbaikan file meta-desc.php.
# Perbaikan component edit theme, dengan menambahkan line number pada textarea.

Build Version 1.1.3 (12 September 2013) :
# Penyempurnaan dalam modul instalasi.
# Penambahan component Menu Manager untuk memudahkan user dalam pembuatan menu dinamis dan telah diterapkan pada theme default chingsy.
# Perbaikan dynamic title pada file title.php.
# Perbaikan dynamic description pada file meta-desc.php.
# Perbaikan bugs pada component category, gallery, pages, post dan user.
# Penambahan field baru pada component user untuk gambar profil dan profil biografi agar di front-end bisa ditampilkan info dari penulis artikel.
# Perbaikan file database PopojiCMS.

Build Version 1.1.2 (24 Agustus 2013) :
# Perubahan besar pada template backend PopojiCMS (admin page), dimana template semula metro diganti dengan template baru universal sehingga bisa dipakai untuk project apapun.
# Component theme telah disempurnakan dengan ditambahkannya edit file theme langsung tanpa perlu masuk ke CPanel lagi untuk mengubah script theme.
# Penambahan One Click and Restore DB, untuk memudahkan user dalam backup dan restore database sehingga tidak perlu menggunakan phpmyadmin lagi.

Build Version 1.1.1 (21 Juli 2013) :
# Penambahan satu field id pada masing-masing tabel yang ada di component post, category dan pages, untuk memudahkan user melihat id suatu content yang akan ditampilkan pada tema.
# Pada component setting, sudah ditambahkan switch editor sehingga user dapat mengganti text editor sesuai keinginannya. Adapun text editor yang disediakan adalah redactor (default), cleditor serta gabungan tinymce dan fckeditor.

Build Version 1.1.0 (08 Juli 2013) :
# Penyempurnaan sistem, bersama diluncurkannya paket develop PopojiCMS.

Build Version 1.0.5 (03 Juli 2013) :
# Perbaikan plugin FB dan Twitter share dikarena plugin yang lama memakan bandwidth yang cukup besar.
# File comment.php diganti nama menjadi po-postcom.php dikarenakan ada hosting yang tidak mengijinkan pemakaian nama comment.php.
# Penambahan script "error_reporting(0);" pada file title.php, meta-desc.php dan meta-key.php dikarenakan terdapat bugs "path disclosures" jika file tersebut dieksekusi.

Build Version 1.0.4 (25 Juni 2013) :
# Perbaikan error pada versi php 5.3.x keatas.

Build Version 1.0.3 (21 Juni 2013) :
# Perbaikan pada file meta-desc.php dan meta-key.php.

Build Version 1.0.2 (10 Juni 2013) :
# Perbaikan statistik untuk traffic web pada halaman frontend.


changelog.md
# Changelog PopojiCMS

### Build Version 2.0.1 (07 Agustus 2016) :
- Perbaikan pemanggilan library utama pada core.php
- Penambahan pilihan timezone pada proses instalasi.
- Penambahan requirement php extention fileinfo pada proses instalasi.
- Penambahan option penggunaan vQmod pada config.php
- Penambahan option timezone pada config.php
- Perbaikan publish date yang bermasalah dengan menyamakan waktu sistem dan database.
- Perbaikan modal window tinymce menjadi responsive.
- Perbaikan masalah pada component library dan filemanager.
- Penambahan rss feed pada frontend.
- Penambahan snipped post populer per kategori.
- Perbaikan engine dynamic menu.
- Perbaikan masalah pengiriman email notifikasi.
- Perbaikan plugin javascript time pada component post.
- Perbaikan component user pada masalah edit data.
- Penambahan option permalink pada url post.
- Penambahan info date dan hits pada component tabel post.
- Perbaikan design halaman login member.
- Penambahan halaman profile member.
- Penambahan halaman detail statistik visitor.

### Build Version 2.0.0 (07 Juni 2016) :
- Perubahan keseluruhan engine dari native ke struktur MVC tetapi tetap dengan rasa native.
- Penambahan tag pada bagian frontend.
- Penambahan komentar sehingga sudah mendukung struktur tree (parent-child).
- Penambahan data konfigurasi persub pada bagian pengaturan engine.
- Pergantian wajah theme chingsy.
- Telah mendukung 100% engine multi bahasa.
- Galeri pada frontend sudah dibuat peralbum.
- Penambahan gambar galeri pada component post.
- Penambahan schedule post.
- Penambahan kategori sehingga sudah mendukung struktur tree (parent-child).
- Setiap post sudah mendukung multi kategori.
- Perbaikan sistem pencarian post pada frontend.
- Perbaikan component pustaka sehingga semua file yang diupload akan termanage dari sini.
- Penambahan konfigurasi pada engine theme.
- Penambahan istilah widget pada component.
- Perbaikan level pengguna yang lebih dinamis.
- Penambahan menu admin yang bisa diganti lewat component menu manager.
- Penambahan engine ternama vqmod.
- Support semua database yang bisa dijalankan dengan PDO.
- Penambahan dashboard member terpisah.
- Penambahan picture description pada post.

### Build Version 1.3.0 (09 Februari 2015) :
- Tahun baru, wajah baru.
- Perbaikan engine menu yang terjadi error double class ketika digenerate.
- Menghilangkan fitur voice recognizer (sementara) yang tidak support lagi di chrome yang baru.
- Update engine filemanager terbaru sesuai update dari pihak ketiga.
- Penyesuain element html dengan W3C Standards.
- Rilis CoGen v.1.0.0 yaitu tool untuk meng-generate component otomatis tanpa perlu menulis script lagi.
- Penambahan social media meta tag sehingga setiap post akan lebih spesifik terbaca ketika share link di social media.
- Perbaikan fungsi extract file zip.
- Penambahan fast menu untuk tambah baru data di component-component.
- Penambahan permalink dan publish date pada component post dan pages.
- Perbaikan filtering post pada back-end sehingga semua user yang berlevel admin dapat melihat semua post dari user yang lain.
- Penambahan action reply pada component contact.
- Penambahan plugin jquery untuk progres form yang ada di login, register, activation dan recover.
- Penambahan tipe login password yaitu login dengan pattern seperti pada android.
- Penambahan image medium untuk setiap image yg diupload (po_upload/medium/medium_namaimage.extensi).
- Penambahan sitemap generator dibagian setting/pengaturan po-admin.
- Pembaharuan template back-end dan frond-end agar bisa dibuka cross browser (termasuk IE).
- Captcha di komentar sudah diganti dengan new reCaptcha dari google.
- Perbaikan maintenance template.
- Penambahan fitur headline dan publish time pada component post.
- Perbaikan pada editor tinymce.
- Perbaikan security yang ada pada tiap component.

### Build Version 1.2.5 (12 Agustus 2014) :
- Perbaikan fitur theme editor menggunakan script Code Mirror, menjadikan editor theme lebih powerfull lagi.
- Penambahan helper pada theme editor, sehingga memudahkan developer untuk membuat themenya sendiri.
- Penambahan fitur Speech Recognition pada component post sehingga menulis tidak perlu lagi mengetikan keyboard tetapi cukup dengan berbicara pada microphone saja (Only Google Chrome).
- Penambahan user registrasi menjadikan PopojiCMS menjadi web multiuser yang dinamis sesuai dengan hak akses tiap user.
- Penambahan component user akses sehingga secara default hak akses terdiri dari admin, user dan member, yang nantinya bisa ditambahkan sendiri hak akses sesuai kebutuhan end user.
- Penambahan component user role dimana component ini dapat mengatur secara dinamis hak akses masing-masing content yang ada di halaman administrator sesuai dengan user akses tertentu.
- Perbaikan .htaccess dimana telah diatur slug url baru yang lebih SEO Friendly lagi, hal ini untuk menanggapi permintaan dari kebanyakan end user yang menginginkan slug url yang lebih baik.
- Perbaikan 2 theme default mengikuti slug url baru serta perbaikan fitur pagging yang lebih dinamis lagi.
- Perbaikan pada title, meta descriptions, meta keywords untuk hasil SEO yang lebih baik lagi.

### Build Version 1.2.4 (7 Juni 2014) :
- Perbaikan pada menu engine, dimana sekarang sudah bisa support multi class sampai 3 class ke bawah.
- Perbaikan pada .htaccess baik pada frond-end ataupun pada back-end.
- Perbaikan setiap inputan pada frond-end tidak lagi menggunakan ajax tetapi melalui pemprosesan php murni, hal ini untuk mempermudah developing theme nantinya.
- Penambahan fitur pagging dimana sekarang bisa dipilih untuk pagging berupa kombinasi number atau hanya tombol prev dan next saja.
- Perbaikan penulisan coding pada komponen galeri dan user.
- Perbaikan pada komponen post dimana sekarang untuk user yang tidak memiliki hak akses admin hanya bisa melihat dan mengedit konten yang ditulis oleh dirinya sendiri.


404.php
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
	<title>404 - Not Found</title>
	<meta charset="utf-8" />
</head>
<body>
	<pre>
          .--.
         / _  \  ___      .--.
        | ( _.-""   `'-.,' _  \
         \.'            '.  ) /
         /                \_.'
        /    .-.   .-.     \
        |   / o \ / o \    |
        ;   \.-'` `'-./    |
        /\      ._.       /
      ;-'';_   ,_Y_,   _.'
     /     \`--.___.--;.
    /|      '.__.---.  \\
   ;  \              \  ;'--. .-.
   |   '.    __..-._.'  |-,.-'  /
   |     `""`  .---._  / .--.  /
  / ;         /      `-;/  /|_/
  \_,\       |            | |
  404 '-...--'._     _..__\/
                `"""`
  So tau nda ada ni url masi le mo buka,
  kena toch...
	</pre>
</body>
</html>

maintenance.php
<?php
session_start();
/**
 * Memanggil library utama
 *
 * Call main library
 *
*/
include_once 'po-includes/core/config.php';

if (VQMOD == TRUE) {
	require_once 'vqmod/vqmod.php';
	VQMod::bootup();
	include_once VQMod::modCheck('po-includes/core/core.php');
	$core = new PoCore();
} else {
	include_once 'po-includes/core/core.php';
	$core = new PoCore();
}

/**
 * Alihkan ke index.php jika mode pemeliharaan tidak aktif
 *
 * Redirect to index.php if maintenance mode not activated
 *
*/
if ($core->posetting[16]['value'] == 'N') {
	header('location:./');
} else {
?>
<!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Strict//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd">
<html xmlns="http://www.w3.org/1999/xhtml" xml:lang="en" lang="en">
<head>
    <meta charset="utf-8" />
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <meta http-equiv="imagetoolbar" content="no" />
    <meta http-equiv="Copyright" content="<?=CONF_STRUCTURE;?>" />
	<meta name="robots" content="index, follow" />
    <meta name="description" content="Log In Panel <?=CONF_STRUCTURE;?>" />
    <meta name="generator" content="<?=CONF_STRUCTURE;?> <?=CONF_VER;?>.<?=CONF_BUILD;?>" />
    <meta name="author" content="Dwira Survivor" />
    <meta name="language" content="Indonesia" />
    <meta name="revisit-after" content="7" />
    <meta name="webcrawlers" content="all" />
    <meta name="rating" content="general" />
    <meta name="spiders" content="all" />
    <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1.0" />
    <!--[if gt IE 8]>
        <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <![endif]-->
    <title>Under Maintenance</title>
    <link rel="shortcut icon" href="<?=DIR_INC;?>/images/favicon.png" />

	<link type="text/css" rel="stylesheet" href="<?=DIR_INC;?>/css/bootstrap.min.css" />
	<link type="text/css" rel="stylesheet" href="<?=DIR_INC;?>/css/font-awesome.min.css" />
	<link type="text/css" rel="stylesheet" href="<?=DIR_INC;?>/css/login.css" />
	<link type="text/css" rel="stylesheet" href="<?=DIR_INC;?>/css/patternlock.css" />

	<script type="text/javascript" src="<?=DIR_INC;?>/js/jquery/jquery-2.1.4.min.js"></script>
	<script type="text/javascript" src="<?=DIR_INC;?>/js/bootstrap/bootstrap.min.js"></script>

	<!-- HTML5 shim and Respond.js for IE8 support of HTML5 elements and media queries -->
    <!--[if lt IE 9]>
      <script src="https://oss.maxcdn.com/html5shiv/3.7.2/html5shiv.min.js"></script>
      <script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
    <![endif]-->

	<style>
		body { background-color: #f1f1f1; text-align: center; padding: 150px 30px; }
		h1 { letter-spacing: -1px; line-height: 60px; font-size: 40px; }
		body { font: 20px "Helvetica Neue", Helvetica, Arial, sans-serif; color: #666666; }
		article { display: block; text-align: left; margin: 0 auto; }
		a { color: #4183c4; text-decoration: none; }
		a:hover { color: #666666; text-decoration: none; }
		.text-small { font-size: 18px; }
	</style>
</head>
<body>
	<article class="col-md-8 col-md-offset-2 text-center">
		<h1>We&rsquo;ll be back soon!</h1>
		<div>
			<p>Sorry for the inconvenience but we&rsquo;re performing some maintenance at the moment. If you need to you can always <a href="mailto:<?=$core->posetting[5]['value'];?>">contact us</a>, otherwise we&rsquo;ll be back online shortly!</p>
			<p class="text-small">&mdash; <?=$core->posetting[0]['value'];?></p>
		</div>
	</article>
</body>
</html>
<?php } ?>

index.php
<?php
/**
 *
 * - PopojiCMS Front End File
 *
 * - File : index.php
 * - Version : 1.2
 * - Author : Jenuar Dalapang
 * - License : MIT License
 *
 *
 * Ini adalah file yang akan di panggil pertama kali ketika mengakes website.
 * This is a file will be called first when accessing the website.
 *
*/

/**
 * Memulai session
 *
 * Start session
 *
*/
session_start();

/**
 * Mendefinisikan Base Url
 *
 * Define Base Url
 *
*/
$base_root = ((isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] && !in_array(strtolower($_SERVER['HTTPS']),array('off','no'))) ? 'https' : 'http').'://'.$_SERVER['HTTP_HOST'].$_SERVER["PHP_SELF"];
$base_url = preg_replace("/\/(index\.php$)/", "", $base_root);
define('BASE_URL', $base_url);

/**
 * Memfilter jika engine belum di install
 *
 * Filtering if engine not installed
 *
*/
if (file_exists('install.php')) {
	header('location:'.BASE_URL.'/install.php');
} else {
	/**
	 * Memanggil library utama
	 *
	 * Call main library
	 *
	*/
	include_once 'po-includes/core/config.php';

	if (VQMOD == TRUE) {
		require_once 'vqmod/vqmod.php';
		VQMod::bootup();
		include_once VQMod::modCheck('po-includes/core/core.php');
	} else {
		include_once 'po-includes/core/core.php';
	}

	/**
	 * Memisahkan request ke bagian admin dan ke bagian depan
	 *
	 * Filtering request to admin and to front end
	 *
	*/
	if (strpos($_SERVER['REQUEST_URI'], DIR_ADM) !== false) {
		header('location:index.php');
	} else {
		/**
		 * Mendeklarasi class PoCore
		 *
		 * Declaration PoCore class
		 *
		*/
		$core = new PoCore();

		/**
		 * Mendeklarasi pemilihan bahasa
		 *
		 * Declaration language choosing
		 *
		 * Add change language with get method (v.2.0.2)
		*/
		if (isset($_POST['lang'])) {
			$check_lang = $core->podb->from('language')
				->where('code', $core->postring->valid($_POST['lang'], 'xss'))
				->limit(1)
				->fetch();
			if ($check_lang) {
				setcookie('po_lang_front', $core->postring->valid($_POST['lang'], 'xss'), 1719241200, '/');
				define('WEB_LANG_ID', $check_lang['id_language']);
				define('WEB_LANG', $check_lang['code']);
			} else {
				$current_lang = $core->podb->from('language')
					->where('id_language', '1')
					->limit(1)
					->fetch();
				define('WEB_LANG_ID', $current_lang['id_language']);
				define('WEB_LANG', $current_lang['code']);
			}
			header('location:'.$_POST['refer']);
		} elseif (isset($_GET['lang'])) {
			$check_lang = $core->podb->from('language')
				->where('code', $core->postring->valid($_GET['lang'], 'xss'))
				->limit(1)
				->fetch();
			if ($check_lang) {
				setcookie('po_lang_front', $core->postring->valid($_GET['lang'], 'xss'), 1719241200, '/');
				define('WEB_LANG_ID', $check_lang['id_language']);
				define('WEB_LANG', $check_lang['code']);
			} else {
				$current_lang = $core->podb->from('language')
					->where('id_language', '1')
					->limit(1)
					->fetch();
				define('WEB_LANG_ID', $current_lang['id_language']);
				define('WEB_LANG', $current_lang['code']);
			}
		} elseif (isset($_COOKIE['po_lang_front'])) {
			$current_lang = $core->podb->from('language')
				->where('code', $_COOKIE['po_lang_front'])
				->limit(1)
				->fetch();
			define('WEB_LANG_ID', $current_lang['id_language']);
			define('WEB_LANG', $current_lang['code']);
		} else {
			$current_lang = $core->podb->from('language')
				->where('id_language', '1')
				->limit(1)
				->fetch();
			define('WEB_LANG_ID', $current_lang['id_language']);
			define('WEB_LANG', $current_lang['code']);
		}

		/**
		 * Statistik pengunjung
		 *
		 * Visitor statistic
		 *
		*/
		$browser = new PoBrowser();
		$ip_stat = $_SERVER['REMOTE_ADDR'];
		$date_stat = date("Ymd");
		$time_stat = time();
		$browser_stat = $browser->getBrowser();
		$os_stat = $browser->getUserAgent();
		$platform_stat = $browser->getPlatform();
		if ($core->porequest->check_internet_connection()) {
			$ip_data = json_decode(@file_get_contents("http://www.geoplugin.net/json.gp?ip=".$ip_stat));    
			if ($ip_data && $ip_data->geoplugin_countryName != null) {
				$country_stat = $ip_data->geoplugin_countryName;
				$city_stat = $ip_data->geoplugin_city;
			} else {
				$country_stat = '';
				$city_stat = '';
			}
		} else {
			$country_stat = '';
			$city_stat = '';
		}

		$statistics = $core->podb->from('traffic')
			->where('ip', $ip_stat)
			->where('country', $country_stat)
			->where('city', $city_stat)
			->where('date', $date_stat)
			->count();
		if ($statistics > 0) {
			$current_statistic = $core->podb->from('traffic')
				->where('ip', $ip_stat)
				->where('country', $country_stat)
				->where('city', $city_stat)
				->where('date', $date_stat)
				->limit(1)
				->fetch();
			$data_stat = array(
				'hits' => $current_statistic['hits']+1,
				'online' => $time_stat
			);
			$query_stat = $core->podb->update('traffic')
				->set($data_stat)
				->where('ip', $ip_stat)
				->where('country', $country_stat)
				->where('city', $city_stat)
				->where('date', $date_stat);
			$query_stat->execute();
		} else {
			$data_stat = array(
				'ip' => $ip_stat,
				'browser' => $browser_stat,
				'os' => $os_stat,
				'platform' => $platform_stat,
				'country' => $country_stat,
				'city' => $city_stat,
				'date' => $date_stat,
				'hits' => 1,
				'online' => $time_stat
			);
			$query_stat = $core->podb->insertInto('traffic')->values($data_stat);
			$query_stat->execute();
		}

		/**
		 * Mendeklarasi class PoRouter
		 *
		 * Declaration PoRouter class
		 *
		*/
		$router = new PoRouter();

		/**
		 * Alihkan permintaan ke 404, jika router tidak ditemukan
		 *
		 * Redirect request to 404, if router not found
		 *
		*/
		$router->set404(function() {
			header('HTTP/1.1 404 Not Found');
			if (VQMOD == TRUE) {
				include_once VQMod::modCheck('404.php');
			} else {
				include_once '404.php';
			}
		});

		/**
		 * Mendeklarasi class Template
		 *
		 * Declaration Template class
		 *
		*/
		if (strpos($_SERVER['REQUEST_URI'], 'member') !== false) {
			$active_template = "/member";
		} else {
			$active_template = "/".$core->potheme['folder'];
		}
		$templates = new PoTemplate\Engine(DIR_CON.'/themes'.$active_template);
		$templates->loadExtension(new PoTemplate\Extension\PoCore());
		$templates->loadExtension(new PoTemplate\Extension\Asset(DIR_CON.'/themes'.$active_template));
		$templates->loadExtension(new PoTemplate\Extension\AssetCssJs());

		/**
		 * Memanggil file komponen untuk bagian depan
		 *
		 * Call component file for front end
		 *
		*/
		$get_components = new PoDirectory();
		$components = $get_components->listDir(DIR_CON.'/component/');
		foreach($components as $component) {
			if ($component != 'index.html') {
				if (file_exists(DIR_CON.'/component/'.$component.'/'.$component.'.php')) {
					if (VQMOD == TRUE) {
						include_once VQMod::modCheck(DIR_CON.'/component/'.$component.'/'.$component.'.php');
					} else {
						include_once DIR_CON.'/component/'.$component.'/'.$component.'.php';
					}
				}
			}
		}

		/**
		 * Memanggil file widget untuk bagian depan
		 *
		 * Call component widget for front end
		 *
		*/
		$get_widgets = new PoDirectory();
		$widgets = $get_widgets->listDir(DIR_CON.'/widget/');
		foreach($widgets as $widget) {
			if ($widget != 'index.html') {
				if (file_exists(DIR_CON.'/widget/'.$widget.'/'.$widget.'.php')) {
					if (VQMOD == TRUE) {
						include_once VQMod::modCheck(DIR_CON.'/widget/'.$widget.'/'.$widget.'.php');
					} else {
						include_once DIR_CON.'/widget/'.$widget.'/'.$widget.'.php';
					}
					$widget_name = ucfirst($widget);
					$templates->loadExtension(new $widget_name());
				}
			}
		}

		/**
		 * Menjalankan router untuk mode pemeliharaan
		 *
		 * Run router for maintenance mode
		 *
		*/
		$router->before('GET|POST', '/.*', function() use ($core) {
			if ($core->posetting[16]['value'] == 'Y') {
				header('location:'.BASE_URL.'/maintenance.php');
				exit();
			}
		});

		/**
		 * Menjalankan router
		 *
		 * Run router
		 *
		*/
		$router->run(function() use ($templates) {
			return $templates;
		});
	}
}